frontend/public/components/monitoring.tsx

```diff
@@ -1750,7 +1751,7 @@ const getAlertsAndRules = (
   });
 
   // Add `rule` object to each alert
-  const alerts = _.flatMap(rules, (rule) => rule.alerts.map((a) => ({ rule, ...a })));
+  const alerts = _.flatMap(rules, (rule) => rule?.alerts?.map((a) => ({ rule, ...a })));
```

Comment:
I think we shouldn't be adding the `?` here since `rules` is an array of `Rule` objects and `alerts` is a required field of `Rule`.

Reply:
In Thanos rule alerts value is ```null``` when there are no alerts are present.  Whereas in the case of Prometheus rule alerts value is ```[]``` when there are no alerts are present.